draft/XXXX-upload-system/README.md

```diff
@@ -0,0 +1,126 @@
+* Fault tolerance and upload pausing/resuming
+* Progress tracking
+
+This would be accomplished through the following additions.
+
+### Hidden Tables/Networks
+To allow for uploaded chunks to be stored somewhere, when starting an upload, the desired table/network would be created in a "hidden" mode. This would prevent the table from being returned by any table listing endpoints. This table should be assumed to be volatile, as data is still being uploaded to it in this state. Whether a table is hidden or not would be tracked by a `_meta` collection on each workspace. This would contain, among other things, the information of any hidden tables in that workspace. Each document in this collection will have a structure like so:
+
+```
+{
+  collection: id,
+  hidden: bool,
+}
+```
+
+The `collection` key would point to the ID of the collection to which this document refers.
+
+Other information could be added to this collection in subsequent RFCs, making this a general purpose collection metadata storage mechanism.
+
+#### Alternate Approach
+Instead of modifying the table/network listing endpoints to omit these "hidden" tables, these endpoints could instead be modified to return objects (dictionaries), which along with the table/networks names, would include if they are temporary (i.e. being actively uploaded to). The benefit to this would be that users wouldn't be able to accidentally create duplicate tables of the same name, as they could see all tables which may exist. However, this would allow for any user (who has access to the workspace) to see an in-progress table, which may contain garbage data, and is likely unvalidated. Furthermore, if an upload fails validation after the fact, it may once again disappear from the list of tables/networks, possibly causing more confusion.
+
+### System-wide Upload Collection
+To keep track of current uploads, a `_system` collection called `uploads` would be created, which would keep track of any ongoing uploads. This collection would store, among other things, the progress of the upload, the "hidden" table(s)/graph(s) to which it belongs, and the user who initiated the upload. The schema for each document in this collection would have a structure like so:
+
+```
+{
+  current: int (bytes),
+  total: int (bytes),
+  done: bool,
+  graph: bool (indicates that a graph is being uploaded),
+  workspace: string,
+  target: {
+    tables: string[],
+    graphs: string[],
+  },
+  user: user_id (sub value),
+  _id: id (arangodb, used by clients to actually send chunks)
+}
+```
+
+### Initializing an Upload
+To initialize an upload, new endpoints will be needed, which will replace the existing upload endpoints. To initialize an upload, a request of the following format is made (endpoint for each upload format specified):
+
+```
+POST /api/upload/csv/{workspace}/{table}
+POST /api/upload/d3_json/{workspace}/{graph}
+POST /api/upload/nested_json/{workspace}/{graph}
+POST /api/upload/newick/{workspace}/{graph}
```

Comment:
Minor (but important) note--the endpoint name should be `uploads` to be consistent with the use of plural in the other REST API sections.